rosbridge_client/src/rosbridge_client/websocket_handler.py

```diff
@@ -0,0 +1,102 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import rospy
+import thread
+import time
+from ws4py.client.threadedclient import WebSocketClient
+
+from rosbridge_library.rosbridge_protocol import RosbridgeProtocol
+from rosbridge_library.util import json, bson
+
+
+class RosbridgeWebSocketClient(WebSocketClient):
+    """
+    Connects to exisiting webserver to wait for incoming data from user.
+
+    This is minimal implementation which may not have full feature.
+    """
+
+    # The following are passed on to RosbridgeProtocol
+    # defragmentation.py:
+    fragment_timeout = 600  # seconds
+    # protocol.py:
+    delay_between_messages = 0  # seconds
+    max_message_size = None  # bytes
+    unregister_timeout = 10.0  # seconds
+    bson_only_mode = False
+    seed = 0
+
+    def opened(self):
+        """
+        Called by the server when the handshake is successful
+        """
+        cls = self.__class__
+        parameters = {
+            "fragment_timeout": cls.fragment_timeout,
+            "delay_between_messages": cls.delay_between_messages,
+            "max_message_size": cls.max_message_size,
+            "unregister_timeout": cls.unregister_timeout,
+            "bson_only_mode": cls.bson_only_mode
+        }
+        try:
+            self.protocol = RosbridgeProtocol(cls.seed, parameters=parameters)
+            self.protocol.outgoing = self.send_message
+            self.authenticated = False
+        except Exception as exc:
```

Comment:
`pass` is not needed here

```suggestion
            except Exception as exc:
                rospy.logerr("Unable to accept incoming connection.  Reason: %s",
                         str(exc))
```